assets/pages/new-contributions-landing/components/StripePaymentRequestButton.jsx

```diff
@@ -0,0 +1,198 @@
+  setCanMakeApplePayPayment: (boolean) => void,
+  setPaymentRequest: (Object) => void,
+  onPaymentAuthorised: (PaymentAuthorisation, Function) => void,
+  setStripePaymentRequestButtonClicked: () => void,
+  isTestUser: boolean,
+  updateEmail: string => void,
+  selectedAmounts: SelectedAmounts,
+  otherAmounts: OtherAmounts,
+  contributionType: ContributionType,
+  countryGroupId: CountryGroupId,
+  amount: number,
+|};
+
+
+const mapStateToProps = (state: State) => ({
+  selectedAmounts: state.page.form.selectedAmounts,
+  otherAmounts: state.page.form.formData.otherAmounts,
+  canMakeApplePayPayment: state.page.form.stripePaymentRequestButtonData.canMakeApplePayPayment,
+  paymentRequest: state.page.form.stripePaymentRequestButtonData.paymentRequest,
+  countryGroupId: state.common.internationalisation.countryGroupId,
+  country: state.common.internationalisation.countryId,
+  currency: state.common.internationalisation.currencyId.toLowerCase(),
+  isTestUser: state.page.user.isTestUser || false,
+  contributionType: state.page.form.contributionType,
+});
+
+const mapDispatchToProps = (dispatch: Function) => ({
+  onPaymentAuthorised:
+    (token, complete) => { dispatch(onStripePaymentRequestApiPaymentAuthorised(token, complete)); },
+  setCanMakeApplePayPayment:
+    (canMakeApplePayPayment) => { dispatch(setCanMakeApplePayPayment(canMakeApplePayPayment)); },
+  setPaymentRequest:
+    (paymentRequest) => { dispatch(setPaymentRequest(paymentRequest)); },
+  updateEmail: (email: string) => { dispatch(updateEmail(email)); },
+  setStripePaymentRequestButtonClicked: () => { dispatch(setStripePaymentRequestButtonClicked()); },
+});
+
+
+// ----- Functions -----//
+
+
+function updateUserEmail(data: Object, setEmail: string => void) {
+  const email = data.payerEmail;
+  if (email) {
+    if (isValidEmail(email)) {
+      setEmail(email);
+    } else {
+      logException(`Failed to set email for stripe payment request user with email: ${email}`);
+    }
+  } else {
+    logException('Failed to set email: no email in data object');
+  }
+}
+
+const onComplete = (complete: Function) => (res: PaymentResult) => {
+  if (res.paymentStatus === 'success') {
+    complete('success');
+  } else if (res.paymentStatus === 'failure') {
+    complete('fail');
+  }
+};
+
+function onClick(event, props: PropTypes) {
+  event.preventDefault();
+  props.setStripePaymentRequestButtonClicked();
+  const amountIsValid =
+    checkAmountOrOtherAmount(
+      props.selectedAmounts,
+      props.otherAmounts,
+      props.contributionType,
+      props.countryGroupId,
+    );
+  if (props.paymentRequest && amountIsValid) {
+    props.paymentRequest.show();
+  }
+}
+
+function initialisePaymentRequest(props: PropTypes) {
+  const paymentRequest = props.stripe.paymentRequest({
+    country: props.country,
+    currency: props.currency,
+    total: {
+      label: 'The Guardian',
+      amount: props.amount,
+    },
+    requestPayerEmail: true,
```

Comment:
so do we hide the email field on the form if they're using the Payment Request button?

Reply:
oh I see, we just say "or". I guess you've probably discussed this at length already but it would be interesting to see how it performs if we just hide all the rest. Maybe as a separate variant/later test